frontend/src/components/GameState/index.tsx

```diff
@@ -2,111 +2,130 @@ import GameAnsweringQuestion from "./GameAnsweringQuestion";
 import GameSelectingQuestion from "./GameSelectingQuestion";
 import GameWaiting from "./GameWaiting";
 import GameOver from "./GameOver";
-
 import { State } from "../../types";
-
-import { useEffect, useState } from "react";
-
+import { useCallback, useEffect, useState } from "react";
 import useSound from "use-sound";
+import { processState } from "../../lib/utils";
 
 interface GameStateProps {
   state: State;
   role: string;
 }
 
 export default function GameState({ state, role }: GameStateProps) {
-  const [startAnimation, setStartAnimation] = useState<boolean>(false);
-  const [animationPosition, setAnimationPosition] = useState<number>(1);
-
+  const [startOpenAnimation, setStartOpenAnimation] = useState<boolean>(false);
+  const [startFadeOutAnimation, setStartFadeOutAnimation] =
+    useState<boolean>(false);
+  const [selectingQuestionInView, setSelectingQuestionInView] =
+    useState<boolean>(false);
+  const [animationPosition, setAnimationPosition] = useState<number>(0);
   const [selectingQuestion, setSelectingQuestion] = useState<boolean>(false);
   const [answeringQuestion, setAnsweringQuestion] = useState<boolean>(false);
+  const [answeringQuestionInView, setAnsweringQuestionInView] =
+    useState<boolean>(false);
   const [gameOver, setGameOver] = useState<boolean>(false);
-  const [waiting, setWaiting] = useState<boolean>(false);  
+  const [waiting, setWaiting] = useState<boolean>(false);
+
+  // const [playThemeSong, { stop }] = useSound("/sounds/themesong.mp3", {
+  //   loop: true,
+  //   volume: 0.2,
+  // });
+
+  // useEffect(() => {
+  //   if (role === "viewer") {
+  //     playThemeSong();
+  //   }
+  // }, [playThemeSong, role]);
```

Comment:
If this won't be used anymore, please delete it

Reply:
It might be, Gustavo is preparing some new sound effects including a song that will be playing in the background, I'll be sure to remove commented junk when it's final.